extensions/script-libraries/drawing/drawing.livecodescript

```diff
@@ -0,0 +1,2960 @@
+	end if
+
+	if tRy is empty then
+		put 0 into tRy
+	end if
+
+	_svgCompileShape xContext, "rect", tX, tY, tWidth, tHeight, tRx, tRy
+end _svgCompileRectangle
+
+private command _svgCompileCircle @xContext, pElement
+	local tCx, tCy, tR
+	put pElement["features"]["cx"] into tCx
+	put pElement["features"]["cy"] into tCy
+	put pElement["features"]["r"] into tR
+	_svgCompileShape xContext, "circle", tCx, tCy, tR
+end _svgCompileCircle
+
+private command _svgCompileEllipse @xContext, pElement
+	local tCx, tCy, tRx, tRy
+	put pElement["features"]["cx"] into tCx
+	put pElement["features"]["cy"] into tCy
+	put pElement["features"]["rx"] into tRx
+	put pElement["features"]["rx"] into tRy
+	_svgCompileShape xContext, "ellipse", tCx, tCy, tRx, tRy
+end _svgCompileEllipse
+
+private command _svgCompileLine @xContext, pElement
+	local tX1, tY1, tX2, tY2
+	put pElement["features"]["x1"] into tX1
+	put pElement["features"]["y1"] into tY1
+	put pElement["features"]["x2"] into tX2
+	put pElement["features"]["y2"] into tY2
+	_svgCompileShape xContext, "line", tX1, tY1, tX2, tY2
+end _svgCompileLine
+
+private command _svgCompilePolyline @xContext, pElement
+	local tPoints
+	put pElement["features"]["points"] into tPoints
+	_svgCompileShape xContext, "polyline", tPoints
+end _svgCompilePolyline
+
+private command _svgCompilePolygon @xContext, pElement
+	local tPoints
+	put pElement["features"]["points"] into tPoints
+	_svgCompileShape xContext, "polygon", tPoints
+end _svgCompilePolygon
+
+private command _svgCompilePath @xContext, pElement
+	local tD
+	put pElement["features"]["d"] into tD
+	_svgCompileShape xContext, "path", tD
+end _svgCompilePath
+
+private command _svgCompileShape @xContext, pType
+	local tOperation
+	put pType into tOperation[1]
+	if param(3) is an array then
+		put param(3) into tOperation[2]
+	else
+		repeat with tIndex = 3 to paramCount()
+			put param(tIndex) into tOperation[2][tIndex - 2]
+		end repeat
+	end if
+
+	_svgCompileState xContext
+
+	_svgContextEmit xContext, tOperation
+end _svgCompileShape
+
+private command _svgCompileState @xContext
+	local tCurrentState, tLastState
+	put _svgContextGetCurrentState(xContext) into tCurrentState
+	put _svgContextGetLastState(xContext) into tLastState
+
+	/* This can be optimized to not emit fill/stroke properties if the fill
+	 * or stroke is none, or the opacity is 0. */
+
+	repeat for each key tStateName in tCurrentState
+		if tCurrentState[tStateName] is tLastState[tStateName] then
+			next repeat
+		end if
+
+		local tOperation
+		put empty into tOperation
+
+		put tStateName into tOperation[1]
+		seqPushOntoBack tOperation, tCurrentState[tStateName]
+
+		_svgContextEmit xContext, tOperation
+	end repeat
+	_svgContextSetLastState xContext, tCurrentState
+end _svgCompileState
+
+/*******************************************************************************
+ *
+ *  SVG ENCODE OPERATIONS
+ *
+ ******************************************************************************/
+
```

Comment:
It's probably worth a note here (and in drawing.cpp) that the order of these things matter and that they should always be identical.

Reply:
Yup